src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -111,7 +111,7 @@ private static (int Index, ServerNode Node) GetPreferredNodeInternal(NodeSelecto
             for (int i = 0; i < len; i++)
             {
                 Debug.Assert(string.IsNullOrEmpty(serverNodes[i].Url) == false, $"Expected serverNodes Url not null or empty but got: \'{serverNodes[i].Url}\'");
-                if (stateFailures[i] == 0)
+                if (stateFailures[i] == 0 && serverNodes[i].ServerRole == ServerNode.Role.Member)
```

Comment:
IIRC this was done for the edge case when the cluster is down, and the only live node is in rehab state... (which can't be promoted to member, because the cluster is down)

Reply:
probably the `UnlikelyEveryoneFaultedChoice` should handle such case